redis/client.py

```diff
@@ -1520,6 +1524,11 @@ def on_connect(self, connection):
             for k, v in self.patterns.items():
                 patterns[self.encoder.decode(k, force=True)] = v
             self.psubscribe(**patterns)
+        if self.shard_channels:
+            shard_channels = {}
+            for k, v in self.shard_channels.items():
```

Comment:
You don't need the if. Since self.shard_channels is already {}, you can just use the loop. 

In fact you should turn this into a dict comprehension and it should run faster.